status.go

```diff
@@ -53,11 +56,13 @@ func (e *Executor) isTaskUpToDate(ctx context.Context, t *taskfile.Task) (bool,
 			return false, err
 		}
 		if !isUpToDate {
-			return false, nil
+			isUpToDateChecker = false
```

Comment:
does `checker.IsUpToDate()` have a side effect of writing the checksum file?

Reply:
indeed

Reply:
https://github.com/go-task/task/blob/d4ed7c3cfc2408bf0978b9d693ff226e3b9c0707/internal/status/checksum.go#L49